metadata/src/test/java/org/apache/kafka/metadata/migration/KRaftMigrationDriverTest.java

```diff
@@ -383,8 +466,8 @@ public ZkMigrationLeadershipState claimControllerLeadership(ZkMigrationLeadershi
             driver.onMetadataUpdate(delta, image, new LogDeltaManifest(provenance,
                 new LeaderAndEpoch(OptionalInt.of(3000), 1), 1, 100, 42));
             Assertions.assertTrue(claimLeaderAttempts.await(1, TimeUnit.MINUTES));
-            TestUtils.waitForCondition(() -> driver.migrationState().get(1, TimeUnit.MINUTES).equals(MigrationDriverState.ZK_MIGRATION),
-                "Waiting for KRaftMigrationDriver to enter ZK_MIGRATION state");
+            TestUtils.waitForCondition(() -> driver.migrationState().get(1, TimeUnit.MINUTES).equals(MigrationDriverState.DUAL_WRITE),
+                "Waiting for KRaftMigrationDriver to enter DUAL_WRITE state");
```

Comment:
This test failed sometimes because `ZK_MIGRATION` is just a middle state, so we might miss to get this state and cause the test failure. Changing to verify the `DUAL_WRITE` state which will be more reliable. This will also verify the `ZK_MIGRATION` state because we can't jump to `DUAL_WRITE` directly.